src/Traits/ArrayTrait.php

```diff
@@ -13,15 +15,19 @@ trait ArrayTrait
      * @param string $rootPrefix
      * @return array<mixed>
      */
-    protected function getFlatArray(array $array, string $delimiter, string $rootPrefix = ''): array
+    protected function getFlatArray(array $array, string $delimiter, string $rootPrefix = '', ?Filter $filter = null): array
     {
         $result = [];
 
         foreach ($array as $key => $value) {
+            if ($filter instanceof Filter && $filter->isKeyAllowed($key) === false) {
+                continue;
```

Comment:
If we don't continue here but save the filtered keys into an additional array, we can remove them during loading and merging those two array together during saving/writing.

@boxblinkracer Would that be a feasible solution for the problem that the filtered keys are remove with the `fix:structure` command?

Reply:
hi
yes thats absolutely fine, a really good idea
so is that now already working, can i merge it like this @matthiashamacher ?

Reply:
No not right now. Wanted your input first so I haven't added it. Will add it later today and ping you so you can merge it.

Reply:
Finally found to time to finalize this pr. Was a little bit more complicated than I thought but I think its an ok solution @boxblinkracer Please have a look.

Reply:
hi @matthiashamacher 
awesome thank you, ill check it out later or this weekend :)
thanks for your work on that, means a lot